com.ibm.streamsx.topology/opt/python/codegen/py_pyTupleTosplTuple.cgt

```diff
@@ -37,7 +37,13 @@ void MY_OPERATOR::fromPythonToPort<%=$oport->getIndex()%>(PyObject *pyTuple <%=$
     if (<%=$ai%> < frs) {
          // Value from the Python function
          PyObject *pyAttrValue = PyTuple_GET_ITEM(pyTuple, <%=$ai%>);
-         if (!SplpyGeneral::isNone(pyAttrValue)) {
+         if (!SplpyGeneral::isNone(pyAttrValue)
+<%
+    if (SPL::CodeGen::Type::isOptional($atype)) {
```

Comment:
Not sure I understand the added code.  If `pyAttrValue` is the SPL NULL then the first condition will be true so the `isSplNull` is short circuited?